--- nginx-orig.rb	2014-05-06 11:45:02.000000000 -0400
+++ nginx-new.rb	2014-05-06 11:45:14.000000000 -0400
@@ -25,10 +25,13 @@
   option 'with-webdav', 'Compile with support for WebDAV module'
   option 'with-debug', 'Compile with support for debug log'
   option 'with-spdy', 'Compile with support for SPDY module'
   option 'with-gunzip', 'Compile with support for gunzip module'
 
+  option 'with-headersmore', 'Compile with support for ngx_headers_more module'
+  option 'with-subs', 'Compile with support for ngx_substitutions_filter module'
+
   depends_on 'pcre'
   depends_on 'passenger' => :optional
   depends_on 'openssl'
 
   def passenger_config_args
@@ -50,10 +53,28 @@
     pcre = Formula["pcre"]
     openssl = Formula["openssl"]
     cc_opt = "-I#{pcre.include} -I#{openssl.include}"
     ld_opt = "-L#{pcre.lib} -L#{openssl.lib}"
 
+    if build.with? 'headersmore'
+      cd '/tmp' do
+        rm_rf "headers-more-nginx-module-0.25"
+        rm_rf "headers-more-nginx-module-0.25.tar.gz"
+        curl '-s', '-o', 'headers-more-nginx-module-0.25.tar.gz', 'https://github.com/agentzh/headers-more-nginx-module/archive/v0.25.tar.gz'
+        system "tar", ['-xzf', 'headers-more-nginx-module-0.25.tar.gz']
+      end
+    end
+
+    if build.with? 'subs'
+      cd '/tmp' do
+        rm_rf "ngx_http_substitutions_filter_module-0.6.4"
+        rm_rf "ngx_http_substitutions_filter_module-0.6.4.tar.gz"
+        curl '-s', '-o', 'ngx_http_substitutions_filter_module-0.6.4.tar.gz', 'https://github.com/yaoweibin/ngx_http_substitutions_filter_module/archive/0.6.4.tar.gz'
+        system "tar", ['-xzf', 'ngx_http_substitutions_filter_module-0.6.4.tar.gz']
+      end
+    end
+
     args = ["--prefix=#{prefix}",
             "--with-http_ssl_module",
             "--with-pcre",
             "--with-ipv6",
             "--sbin-path=#{bin}/nginx",
@@ -76,19 +97,29 @@
     args << "--with-http_dav_module" if build.with? "webdav"
     args << "--with-debug" if build.with? "debug"
     args << "--with-http_spdy_module" if build.with? "spdy"
     args << "--with-http_gunzip_module" if build.with? "gunzip"
 
+    args << "--add-module=/tmp/headers-more-nginx-module-0.25" if build.include? 'with-headersmore'
+    args << "--add-module=/tmp/ngx_http_substitutions_filter_module-0.6.4" if build.include? 'with-subs'
+
     if build.head?
       system "./auto/configure", *args
     else
       system "./configure", *args
     end
     system "make"
     system "make install"
     man8.install "objs/nginx.8"
     (var/'run/nginx').mkpath
+
+    if build.with? 'headersmore'
+      rm_rf "/tmp/headers-more-nginx-module-0.21"
+    end
+    if build.with? 'subs'
+      rm_rf "/tmp/ngx_http_substitutions_filter_module-0.6.2"
+    end
   end
 
   def post_install
     # nginx's docroot is #{prefix}/html, this isn't useful, so we symlink it
     # to #{HOMEBREW_PREFIX}/var/www. The reason we symlink instead of patching
